plugins/+ciplugins/+github/BuildSummaryPlugin.m

```diff
@@ -0,0 +1,29 @@
+classdef BuildSummaryPlugin < matlab.buildtool.plugins.BuildRunnerPlugin
+
+%   Copyright 2024 The MathWorks, Inc.
+
+    methods (Access=protected)
+
+        function runTaskGraph(plugin, pluginData)
+            runTaskGraph@matlab.buildtool.plugins.BuildRunnerPlugin(plugin, pluginData);
+            [fID, msg] = fopen(fullfile(getenv("RUNNER_TEMP") ,"buildSummary_" + getenv("GITHUB_RUN_ID") + ".json"), "w");
+
+            if fID == -1
+                warning("ciplugins:github:BuildSummaryPlugin:UnableToOpenFile","Could not open a file for GitHub build summary table due to: %s", msg);
```

Comment:
@mw-hrastega should take a look at these messages.

Reply:
To review the message, I first need to see examples of the string that replaces %s.

Reply:
%s is the message if the MATLAB could not open a specified file for whatever reasons. for example 

<img width="454" alt="image" src="https://github.com/matlab-actions/run-build/assets/47204011/79ce402e-6633-47a0-92ce-9cc90e2fcfd7">

Reply:
Thanks for the example, @nbhoski . Does it matter to the user what the filename is? "A file" sounds like implementation detail.

Reply:
It might be helpful to clarify what the benefit of this warning is, but here is an alternative:

Unable to write the table of MATLAB build results to a file. (%s)

Reply:
> Unable to write the table of MATLAB build results to a file.

It would make debugging easy. in case of we did not see the summary table in the summary section.

Reply:
Ok, I think this message shows the purpose of the file a little better:

"Unable to open a file required to create the MATLAB build summary table. (%s)"

Reply:
Done.

Reply:
Done.